components/table/test/table.vdiff.js

```diff
@@ -0,0 +1,407 @@
+			<th><d2l-table-col-sort-button desc>Descending</d2l-table-col-sort-button></th>
+			<th><d2l-table-col-sort-button nosort>No Sort</d2l-table-col-sort-button></th>
+		</tr>
+	`;
+}
+function createFruitHeaderRows(opts) {
+	const { selectable, headerAttribute, stickyAttribute, stickyClass, trClass } = { selectable: false, headerAttribute: false, stickyAttribute: false, ...opts };
+	return html`
+		<tr class="${ifDefined(trClass)}" ?header="${headerAttribute}">
+			${selectable ? html`<th rowspan="2" style="width: 1%" class="top"><input type="checkbox"></th>` : nothing}
+			<th class="${ifDefined(!selectable ? 'top' : stickyClass)}" ?sticky="${stickyAttribute}" rowspan="2">Country</th>
+			<th colspan="5">Fruit Production (tons)</th>
+		</tr>
+		<tr class="${ifDefined(trClass)}" ?header="${headerAttribute}">
+			<th>Apples</th>
+			<th>Oranges</th>
+			<th>Bananas</th>
+			<th>Peaches</th>
+			<th>Grapes</th>
+		</tr>
+	`;
+}
+
+function createRows(keys, opts) {
+	const { selected } = { selected: false, ...opts };
+	return html`${keys.map(key => html`
+		<tr ?selected="${selected}">
+			<td>${`Cell ${key}-A`}</td>
+			<td>${`Cell ${key}-B`}</td>
+			<td>${`Cell ${key}-C`}</td>
+		</tr>
+	`)}`;
+}
+function createFruitRows(opts) {
+	const { inputNumber, selectable, selected, stickyAttribute, stickyClass } = { inputNumber: true, selectable: false, selected: [false, false, false], stickyAttribute: false, ...opts };
+	return html`
+		<tr ?selected="${selectable && selected[0]}">
+			${selectable ? html`<td class="down"><input type="checkbox" ?checked="${selected[0]}"></td>` : nothing}
+			<th class="${ifDefined(!selectable ? 'down' : stickyClass)}" ?sticky="${stickyAttribute}">Canada</th>
+			<td>356,863</td>
+			<td>0</td>
+			<th>0</th>
+			<td>23,239</td>
+			<td class="over">90,911</td>
+		</tr>
+		<tr ?selected="${selectable && selected[1]}">
+			${selectable ? html`<td><input type="checkbox" ?checked="${selected[1]}"></td>` : nothing}
+			<th class="${ifDefined(stickyClass)}" ?sticky="${stickyAttribute}">Australia</th>
+			${stickyAttribute || stickyClass ? html`<td><d2l-input-number label="label" label-hidden value="308298"></d2l-input-number></td>` : html`<td>308,298</td>`}
+			<td>398,610</td>
+			<td>${inputNumber ? html`<d2l-input-number label="label" label-hidden value="354241"></d2l-input-number>` : '354,241'}</td>
+			<td>80,807</td>
+			<td>1,772,911</td>
+		</tr>
+		<tr ?selected="${selectable && selected[2]}">
+			${selectable ? html`<td><input type="checkbox" ?checked="${selected[2]}"></td>` : nothing}
+			<th class="${ifDefined(stickyClass)}" ?sticky="${stickyAttribute}">Mexico</th>
+			<td>716,931</td>
+			<td>4,603,253</td>
+			<td>2,384,778</td>
+			<td>176,909</td>
+			<td>351,310</td>
+		</tr>
+	`;
+}
+
+describe('table', () => {
+
+	[
+		{ type: 'default', rtl: false  },
+		{ type: 'default', rtl: true  },
+		{ type: 'light', rtl: false  },
+		{ type: 'light', rtl: true }
+	].forEach(({ type, rtl }) => {
```

Comment:
I went with this pattern to try to reduce the amount of nesting, saves us a couple levels 🤷